crates/subspace-networking/src/node_runner.rs

```diff
@@ -584,16 +584,23 @@ where
                     kademlia.add_address(&peer_id, address);
                 }
             } else {
-                trace!(
+                info!(
                     %local_peer_id,
                     %peer_id,
-                    "Peer doesn't support our Kademlia DHT protocol ({:?}). Adding to the DHT skipped.",
+                    peer_protocols=?info.protocols,
+                    "Peer doesn't support our Kademlia DHT protocol ({:?}). Peer was banned.",
                     kademlia
                         .protocol_names()
                         .iter()
                         .map(|p| String::from_utf8_lossy(p.as_ref()))
                         .collect::<Vec<_>>(),
-                )
+                );
+
+                kademlia.remove_peer(&peer_id);
+                self.networking_parameters_registry
+                    .remove_all_known_peer_addresses(peer_id)
+                    .await;
+                self.swarm.ban_peer_id(peer_id);
```

Comment:
This is concerning, it'll not allow us to converge with IPFS network in the future. How do you plan to handle that?

I don't think banning is an appropriate action here. I think it is already sufficient that nodes will not be added to k-buckets of Kademlia and returned from bootstrap nodes.

Reply:
I don't have the requirements for IPFS compatibility yet, but at the first glance - I would change the "kademlia detection" algorithm. If it IPFS-like protocol name - then do X if it's "similar to our network but with a different genesis-hash" - then do Y. 

I would keep bans until we have a strong argument against it because it should reduce the additional work a lot for peers when connected by "alien peers" (establishing 2 connections, running requests prior to "our verdict", keeping established connections, allowing reconnections, and repetition all of the above many times, etc) - bans handle all of that.